pkg/prometheus/promcfg.go

```diff
@@ -2317,6 +2319,24 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 
 	if sc.Spec.RelabelConfigs != nil {
```

Comment:
(nit) not part of this change but checking the length of the slice would be more idiomatic.
```suggestion
	if len(sc.Spec.RelabelConfigs) > 0 {
```